exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -914,8 +1210,49 @@ func (ku *Kucoin) GetOrderInfo(ctx context.Context, orderID string, pair currenc
 			RemainingAmount: orderDetail.Size - orderDetail.DealSize,
 			Amount:          orderDetail.Size,
 			Price:           orderDetail.Price,
-			Date:            orderDetail.CreatedAt.Time()}, nil
+			Date:            orderDetail.CreatedAt.Time(),
+		}, nil
 	case asset.Spot, asset.Margin:
+		if ku.HFMargin || ku.HFSpot {
+			var orderDetail *HFOrderDetail
+			if ku.HFMargin && assetType == asset.Margin {
+				orderDetail, err = ku.GetMarginHFOrderDetailByOrderID(ctx, orderID, pair.String())
+			} else if ku.HFSpot && assetType == asset.Spot {
+				orderDetail, err = ku.GetHFOrderDetailsByOrderID(ctx, orderID, pair.String())
+			}
+			if err != nil {
+				return nil, err
+			}
+			var nPair currency.Pair
+			nPair, err = ku.MatchSymbolWithAvailablePairs(orderDetail.Symbol, assetType, true)
+			if err != nil {
+				return nil, err
+			}
+			oType, err := order.StringToOrderType(orderDetail.OrderType)
+			if err != nil {
+				return nil, err
+			}
+			side, err := order.StringToOrderSide(orderDetail.Side)
+			if err != nil {
+				return nil, err
+			}
+			if !pair.IsEmpty() && !nPair.Equal(pair) {
+				return nil, fmt.Errorf("order with id %s and currency Pair %v does not exist", orderID, pair)
+			}
+			return &order.Detail{
```

Comment:
This looks like you could fill out a lot more details here with the response type:
https://www.kucoin.com/docs/rest/spot-trading/spot-hf-trade-pro-account/get-hf-order-details-by-orderid